From a7c107e09f3e07bbc933da4c2e57ba567d45dec7 Mon Sep 17 00:00:00 2001
Message-Id: <a7c107e09f3e07bbc933da4c2e57ba567d45dec7.1440631943.git.yunwei.zhang@intel.com>
In-Reply-To: <4706a96fcf04085249a994498a8e805f04c2d48e.1440631943.git.yunwei.zhang@intel.com>
References: <4706a96fcf04085249a994498a8e805f04c2d48e.1440631943.git.yunwei.zhang@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Fri, 31 Jul 2015 14:36:43 +0530
Subject: [PATCH 2/2] REVERTME: [VPG]: drm/i915: Fix locking around
 'psr.enabled'

This patch fixes the following behaviors:
1. Access to 'psr.enabled' variable should be protected by
   'psr.lock', to avoid races from PSR enable/disable sequence.
2. Also, 'psr.enabled' will never be set without setup_done set
   to true. So, checking for both conditions is not required in
   exit sequence.

This patch is REVERTME because CHV PSR design in upstream is
different and thus this patch will not be required when
we move to 4.0+ kernel.

Change-Id: I65d374de49a56a5e5d5e6c21ff53ef76fbcce6e7
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-15168
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 5f9f2b6..b846dd9 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2370,9 +2370,8 @@ void intel_vlv_edp_psr_update(struct drm_device *dev)
 void intel_vlv_edp_psr_exit(struct drm_device *dev, bool disable)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
-	struct intel_dp *intel_dp = dev_priv->psr.enabled;
 
-	if (!intel_dp || !intel_dp->psr_setup_done || !IS_VALLEYVIEW(dev))
+	if (!IS_VALLEYVIEW(dev) || !is_edp_psr(dev))
 		return;
 
 	mutex_lock(&dev_priv->psr.lock);
-- 
1.9.1

